agora/Agora/Record.hs

```diff
@@ -0,0 +1,108 @@
+{- |
```

Comment:
Would this module be suitable for up-streaming?

Reply:
Yeah, we could definitely put this somewhere in plutarch-extra.